### ISSUE TITLE
The steps outside 'specs' dir are recognised as duplicate implementation (python)

### ISSUE THREAD

**Issue by jensakejohansson** [getgauge/gauge-vscode#373]
If 'gauge_specs_dir' and 'STEP_IMPL_DIR' points to different directories VSC marks my steps as 'Duplicate step implementation'. It looks good when I open the project folder, but as soon as I start navigate to my steps from the specs-level everything gets marked as duplicates.

If I move my steps so 'gauge_specs_dir' and 'STEP_IMPL_DIR' is the same directory it works as expected.

Gauge version: 1.0.5
Commit Hash: 562f036

Plugins
-------
html-report (4.0.8)
json-report (0.2.2)
python (0.3.5)
screenshot (0.0.1)
spectacle (0.1.3)

**Comment by negiDharmendra** [comment]
@jensakejohansson I have tried with the same setup you have mentioned. 

It would be a great help if you could share a project where this issue is reproducible.

**Comment by jensakejohansson** [comment]
I will see if I can do that later (can not share my working project). However, I think the same issue was with Gauge/Java: see this:
https://github.com/getgauge/gauge-vscode/issues/346

**Comment by NivedhaSenthil** [comment]
@jensakejohansson  #346 and this issue are unrelated.. validation of steps depends on the language runner and not the extension. Do you see the error only in VScode or it fails even when executing from command line ? 

can you please try `gauge validate` and see if you find any clue from it ?

**Comment by jensakejohansson** [comment]
Only problem in VSC, the tests run perfectly fine, both from command line and in VSC. It's just that all the steps gets highlighted in red. ```Gauge validate``` do not report any errors. It's not so good that the VSC-plugin does tell you where it thinks these duplicates exists when you hover over the red highlighting. It should indicate which files are condidered. I have created a minimal project that recreates the issue (just did some folder changes after ```gauge init python```). I'll see if I can share it somehow.

**Comment by jensakejohansson** [comment]
You should be able to see the problem if you download this project:
https://github.com/jensakejohansson/gauge-duplicate
Looks fine when you first open it in VSC. But if you go to the example.spec file and then press ```F12``` to navigate to a step implementation and then go back to the .spec-file again then all steps are marked in red.

**Comment by NivedhaSenthil** [comment]
Given project works fine for me with mentioned versions just had to use python3 since `__init.py__` is not in modules, I have tried on MacOS Mojave which OS you are in ?

**Comment by jensakejohansson** [comment]
Ok. I'm running Windows 10... We are a few people in my team who all have the same issue, so it's not a 'just on my machine' issue ;)

**Comment by shubhamsc** [comment]
@jensakejohansson 
I'm able to replicate the issue in your given project. It's because of windows path you're using
```
gauge_specs_dir = applications/app1
STEP_IMPL_DIR = fwk/steps
```
which is escaping the first character after `/` when work with vscode for python. 

For now you can use windows specific path separator `\\` in your env properties like -
```
STEP_IMPL_DIR = impl_dir\\tests
gauge_specs_dir = src\\specs
```

we'll fix it soon.

**Comment by jensakejohansson** [comment]
Great! I'm developing on a Windows machine, but the project is executed in a pipeline that maybe either be running on a Windows or a Linux machine, so I'm waiting for a fix so we can keep it OS-independent. 
Thanks!

**Comment by negiDharmendra** [comment]
The fix is available in [gauge-python-0.3.6.nightly-2019-05-24](https://bintray.com/gauge/gauge-python/Nightly/0.3.6.nightly-2019-05-24)

**Comment by Debashis9012** [comment]
This issue has been verified and found fixed with the version ``` nightly-2019-05-24```
